src/app/components/dialogs/create-quiz-dialog/create-quiz-dialog.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { QuizService } from '@services/quiz.service';
+import { QuizCategory } from '@shared/interfaces/quizCategory.interface';
+
+@Component({
+  selector: 'app-create-quiz-dialog',
+  templateUrl: './create-quiz-dialog.component.html',
+  styleUrls: ['./create-quiz-dialog.component.scss']
+})
+export class CreateQuizDialogComponent {
+
+  public quizCategories!: QuizCategory[];
+  public categoryId!: string;
+  public quizDifficulty!: string;
+  public pointsPerQuestion!: string;
+  public questionCount!: number;
+  public maxQuestionCount!: number;
+
+  constructor(
+    private dialogRef: MatDialogRef<CreateQuizDialogComponent>,
+    private quizService: QuizService
+  ) {
+    this.quizService.getQuizCategories().subscribe(categories => {
```

Comment:
@KovalivNataliia when use subscriptions, we need to handle unsubscribe to prevent memory leaks. 
Please add unsubscribe handling in OnDestroy